src/main/java/com/thebois/models/world/World.java

```diff
@@ -188,12 +181,66 @@ public IRoleAllocator getRoleAllocator() {
         return colony;
     }
 
+    /**
+     * Returns the IStructureMatrix as a collection of IStructure.
+     *
+     * @return IStructure collection
+     */
+    public Collection<IStructure> getStructureCollection() {
+        return MatrixUtils.matrixToCollection(this.structureMatrix);
+    }
+
     /**
      * Updates the state of the world.
      */
     public void update() {
         colony.update();
     }
 
+    @Override
+    public Optional<IStructure> findNearestStructure(final Position position) {
+        final int searchRow = Math.round(position.getPosY());
+        final int searchCol = Math.round(position.getPosX());
+
+        final int maxSearchRadius = this.worldSize / 2;
```

Comment:
Having half the worldsize will lead to scenarios where they are at the edges and cant find anything right? Seems like that would cause a lot of frustration for the player.